src/tour-of-beers/edit/beer-edit-form.component.ts

```diff
@@ -0,0 +1,66 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import CountryService from '../common/country.service';
+import BeerService from '../beer/beer.service';
+import { Beer } from '../beer/beer.model';
+import { Country } from '../common/country.model';
+import { Location } from '@angular/common';
+
+@Component({
+    selector: 'beer-edit-form',
+    templateUrl: './beer-edit-form.component.html',
+    styleUrls: ['./beer-edit-form.component.css']
+})
+export default class BeerEditFormComponent implements OnInit {
+
+    @Input() beerProfile: Beer;
+
+    private beerToEdit: Beer;
+    private countryList: Country[];
+    private beerEditFormGroup: FormGroup;
+
+    constructor(private formBuilder:FormBuilder, private countryService: CountryService, 
+        private beerService: BeerService, private location: Location) { 
+        this.countryList = this.countryService.getAvailableCountries();
+    }
+
+    public ngOnInit(): void {
+        if (this.isNewBeerProfile()) {
+            this.beerToEdit = this.beerService.createBeer();
+        } else {
+            this.beerToEdit = this.beerProfile;
+        }
+        this.beerEditFormGroup  = this.createFormGroup(this.beerToEdit);
+    }
+
+    private isNewBeerProfile (): boolean {
+        return !this.beerProfile;
+    }
+
+    private createFormGroup (beer: Beer): FormGroup {
+        return this.formBuilder.group({
+            name: [beer.name, [Validators.required]],
+            alcoholPercentage: [beer.alcoholPercentage, [Validators.pattern(/^[1-9]?[0-9]\.\d{1}$/)]],
```

Comment:
if you have this validation logic, maybe better remove the up/down arrow in the form or make it increment/decrement by 0.1